core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -290,7 +295,7 @@ static class V1Writer extends ManifestWriter<DataFile> {
     private final V1Metadata.IndexedManifestEntry entryWrapper;
 
     V1Writer(PartitionSpec spec, OutputFile file, Long snapshotId) {
-      super(spec, file, snapshotId);
+      super(spec, EncryptedFiles.plainAsEncryptedOutput(file), snapshotId);
```

Comment:
todo: accept EncryptedOutputFile, and throw exception if key metadata is not null / empty.